File: frontend/src/app/dtos/meeting/meeting.dto.ts
import { IsNotEmpty, IsString } from 'class-validator';

export class MeetingDto {
  @IsNotEmpty()
  @IsString()
  title: string;

  @IsString()
  description: string;

  meeting_date: string;

  constructor(data: any) {
    this.title = data.title;
    this.description = data.description;
    this.meeting_date = data.meeting_date;
  }
}

File: backend/src/main/java/com/example/ecomanager/dtos/MeetingDTO.java
package com.example.ecomanager.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class MeetingDTO {

    @NotBlank(message = "Title is required!")
    private String title;

    @JsonProperty("meeting_date")
    private LocalDate meetingDate;

    private String description;

}
